scripts/openstack-update-services.sh

```diff
@@ -0,0 +1,160 @@
+    sed -i $OUTFILE -e "s|value: $OPENSTACK_DEPLOYED_VERSION|value: $OPENSTACK_VERSION|"
+
+    update_event Applying Fake Update CR
+
+    oc apply -f $OUTFILE
+fi
+
+oc project $NAMESPACE
+# wait until openstackVersion cr completes reconcile, status.availableVersion should be the same as VERSION
+oc wait $OPENSTACK_VERSION_CR --for=jsonpath='{.status.availableVersion}'=$OPENSTACK_VERSION --timeout=$TIMEOUT
+
+
+OPENSTACK_DEPLOYED_VERSION=$(oc get $OPENSTACK_VERSION_CR --template={{.status.deployedVersion}})
+
+cat <<EOF >openstackversionpatch.yaml
+    "spec": {
+      "targetVersion": "$OPENSTACK_VERSION"
+      }
+EOF
+
+update_event Patching the Openstack Version
+
+oc patch $OPENSTACK_VERSION_CR  --type=merge  --patch-file openstackversionpatch.yaml
+
+# wait for ovn update on control plane
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeout=$TIMEOUT
+
+update_event MinorUpdateOVNControlplane Completed
+
+# start ovn update on data plane
+nodes_with_ovn=()
+# Get the names of all OpenStackDataPlaneNodeSet resources
+openstackdataplanenodesets=$(oc get openstackdataplanenodeset -o custom-columns=NAME:.metadata.name,SERVICES:.spec.services --no-headers)
+
+# Loop through each OpenStackDataPlaneNodeSet
+while read -r node_name services; do
+    # Check if 'ovn' is in the list of services
+    for service in ${services[@]};do
+        if [[ "$service" == *"ovn"* ]]; then
+            nodes_with_ovn+=("- $node_name")
+            break
+        fi
+    done
+done <<< $openstackdataplanenodesets
+
+DATAPLANE_DEPLOYMENT=edpm
+OVN_NODE_SETS=$(printf '    %s\n' "${nodes_with_ovn[@]}")
+
+cat <<EOF >edpm-deployment-ovn-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-ovn-update
+spec:
+  nodeSets:
+$OVN_NODE_SETS
+  servicesOverride:
+    - ovn
+EOF
+
+update_event Applying the OVN CRD
+
+oc create -f edpm-deployment-ovn-update.yaml
+
+oc get openstackdataplanedeployment ${DATAPLANE_DEPLOYMENT}-ovn-update -o yaml
+# wait for ovn dataplane completes
+oc wait $OPENSTACK_VERSION_CR  --for=condition=MinorUpdateOVNDataplane --timeout=$TIMEOUT
+echo "MinorUpdateOVNDataplane completed"
+
+update_event MinorUpdateOVNDataplane Completed
+
+# wait for control plane update to complete
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateControlplane --timeout=$TIMEOUT
+
+update_event MinorUpdateControlplane Completed
+echo "MinorUpdateControlplane completed"
+
+# start data plane plane update for rest of edpm services
+DATAPLANE_NODESETS=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print "    - "  $2}')
+
+cat <<EOF >edpm-deployment-update-services.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-update-services
+spec:
+  nodeSets:
+$DATAPLANE_NODESETS
+  servicesOverride:
+    - update-services
+EOF
+
+update_event Applying the UPDATE-SERVICES CRD
```

Comment:
here as well, should this be `Applying the EDPM UPDATE-SERVICES update deployment CR ? its not applying a CRD

Reply:
also fixed